sympy/integrals/heurisch.py

```diff
@@ -611,10 +611,11 @@ def _integrate(field=None):
         x, y = symbols('x y', cls=Wild, exclude=[I])
         for poly in list(irreducibles):
             m = poly.match(x + I*y)
```

Comment:
or using @jksuom suggestion, maybe something like this:
```
ip = collect(poly, I)
if ip:
    rp = poly.subs(I, 0)
    if rp != poly - expand_mul(I*ip):
        raise NotImplementedError
    pairs.add(rp, ip)
    irreducibles.remove(poly)
```

Reply:
It seems that `collect(poly, I, evaluate=False)` should be used to separate terms involving `I`.

Reply:
So thus?
```
m = collect(poly, I, evaluate=False)
y = m.get(I, 0)
if y:
    x = m.get(1, 0)
    if x + I*y != poly:
        raise NotImplementedError
    pairs.add((x, y))
    irreducibles.remove(poly)
```

Reply:
I'm not sure if the numbers should be SymPy objects (like `m.get(S.One, S.Zero)`). Maybe `continue` should suffice if `x + I*y != poly` (is that even possible?). But else this looks promising.

Reply:
modified